scripts/pre_push_hook.py

```diff
@@ -410,6 +411,38 @@ def does_diff_include_travis_yml_or_js_files(diff_files):
     return False
 
 
+def check_for_backend_python_library_inconsistencies():
+    """Checks the state of the 'third_party/python_libs' folder and compares it
+    to the required libraries specified in 'requirements.txt'.
+    If any inconsistencies are found, the script displays the inconsistencies
+    and exits.
+    """
+    mismatches = install_backend_python_libs.get_mismatches()
+
+    if mismatches:
+        python_utils.PRINT(
+            'Your currently installed python libraries do not match the\n'
+            'libraries listed in your "requirements.txt" file. Here is a\n'
+            'full list of library/version discrepancies:\n')
+
+        python_utils.PRINT(
+            '{:<35} |{:<25}|{:<25}'.format(
+                'Library', 'Requirements Version',
+                'Currently Installed Version'))
+        for library_name, version_strings in mismatches.items():
+            python_utils.PRINT('{:<35} |{:<25}|{:<25}'.format(
+                library_name, version_strings[0], version_strings[1]))
+        python_utils.PRINT('\n')
+        common.print_each_string_after_two_new_lines([
+            'Please fix these discrepancies by editing the `requirements.in`\n'
```

Comment:
What can lead to a situation where the contributor has to edit `requirements.in` file? [I'm unable to think of a situation.]

I was thinking of directly running `install_third_party` instead of asking the contributor if the user never has to edit the requirements.in file.

Reply:
@DubeySandeep probably if the developer has made a mistake and somehow installed the deps manually by themselves?

Reply:
I'm not sure whether manually installing lib will be stored in third_party/python_libs/ dir. Anyway, that won't be resolved by editing requirements.in file, right?

@kevjumba, Assume we have a lib Av1.2 and on develop we updated the same lib to Av1.3, so after installing the new lib in local repo the thrid_party/python_libs/ will not have Av1.2 right? (just wanted to know for my own knowledge)

Reply:
Yes, the new lib will be not be in the directory.